vernac/declare.ml

```diff
@@ -2464,40 +2463,33 @@ let add_mutual_definitions l ~pm ~info ?obl_hook ~uctx
   in
   pm
 
-let admit_prog ~pm prg =
-  let {obls; remaining} = Internal.get_obligations prg in
-  let obls = Array.copy obls in
-    Array.iteri
-      (fun i x ->
-        match x.obl_body with
-        | None ->
-            let x = subst_deps_obl obls x in
-            let uctx = Internal.get_uctx prg in
-            let univs = UState.univ_entry ~poly:false uctx in
-            let kn = declare_constant ~name:x.obl_name ~local:Locality.ImportNeedQualified
-              (ParameterEntry (None, (x.obl_type, univs), None)) ~kind:Decls.(IsAssumption Conjectural)
-            in
-              assumption_message x.obl_name;
-              obls.(i) <- Obligation.set_body ~body:(DefinedObl (kn, Univ.Instance.empty)) x
-        | Some _ -> ())
-      obls;
-  Obls_.update_obls ~pm prg obls 0
-
-(* get_any_prog *)
+let rec admit_prog ~pm prg =
+  let {obls} = Internal.get_obligations prg in
+  let is_open _ x = Option.is_empty x.obl_body && List.is_empty (deps_remaining obls x.obl_deps) in
+  let i = match Array.findi is_open obls with
+    | Some i -> i
+    | None -> CErrors.anomaly (Pp.str "Could not find a solvable obligation.")
+  in
+  let proof = solve_obligation prg i None in
```

Comment:
This could maybe be factored with next_obligation a bit.

Reply:
Does this mean that if the obligation tactic is `fail`, then admitting the obligations will fail?  Is this expected/desired?  (Is it a regression?) Is it documented?  Is it tested?